src/System.Management.Automation/help/HelpFileHelpProvider.cs

```diff
@@ -368,7 +368,7 @@ internal Collection<string> GetExtendedSearchPaths()
                     {
                         // Get all the directories under the module path
                         // * and SearchOption.AllDirectories gets all the version directories.
-                        string[] directories = Directory.GetDirectories(psModulePath, "*", SearchOption.AllDirectories);
+                        IEnumerable<string> directories = Directory.EnumerateDirectories(psModulePath, "*", SearchOption.AllDirectories);
 
                         var possibleModuleDirectories = directories.Where(directory => !ModuleUtils.IsPossibleResourceDirectory(directory));
```

Comment:
We could exclude Linq at all. Example https://source.dot.net/#System.IO.FileSystem/System/IO/Enumeration/FileSystemEnumerableFactory.cs,185

Is the code covered by tests?

Reply:
We should not exclude Linq simply for the sake of it.

We could avoid a delegate allocation by moving the predicate from the `Where` method into an `if` statement inside the `foreach`. I do not see the benefit of replacing `Any` method with procedural code.

We could consider using `FileSystemEnumerable` generally if it was proven to be beneficial, but that is outside of the scope of this PR.